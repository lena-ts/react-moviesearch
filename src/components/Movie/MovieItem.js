import React from 'react';
import PropTypes from 'prop-types';

const MovieItem = ({ id, img, title, genre, release_date, onClick }) => {
  const handleClick = () => {
    onClick(id, genre);
    window.scrollTo(0, 0);
  };

  return (
    <div className="movie-item" onClick={handleClick}>
      <img src={img} />
      <div className="title">
        <div>
          <h4>{title}</h4>
          <span>{genre}</span>
        </div>
        <div className="date">{release_date}</div>
      </div>
    </div>
  );
};

MovieItem.propTypes = {
  img: PropTypes.string,
  title: PropTypes.string,
  genre: PropTypes.string,
  release_date: PropTypes.string,
  id: PropTypes.number,
  onClick: PropTypes.func,
};

export default MovieItem;
