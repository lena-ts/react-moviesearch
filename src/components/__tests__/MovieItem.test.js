import React from 'react';
import MovieItem from '../Movie/MovieItem';
import renderer from 'react-test-renderer';

it('Movie item shapshot', () => {
  const MovieSnpashot = renderer.create(<MovieItem />).toJSON();
  expect(MovieSnpashot).toMatchSnapshot();
});
