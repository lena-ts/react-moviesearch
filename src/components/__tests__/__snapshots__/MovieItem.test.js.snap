// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie item shapshot 1`] = `
<div
  className="movie-item"
  onClick={[Function]}
>
  <img />
  <div
    className="title"
  >
    <div>
      <h4 />
      <span />
    </div>
    <div
      className="date"
    />
  </div>
</div>
`;
